session_0/word_processing.rb

```diff
@@ -2,13 +2,20 @@
 # any non-lower case alphabet (A..Z) to corresponding lower case
 # alphabet
 def lower_case(words)
-  raise NotImplementedError # TODO
+	wr=[]
+	words.each do |w|
+		wr.push(w.downcase)
+	end
+	return wr
 end
 
 # Similar to `lower_case`, this function modifies the array in-place
 # and does not return any value.
 def lower_case!(words)
-  raise NotImplementedError # TODO
+  words.map do |w|
+    w.downcase!
+   end
+  return words
```

Comment:
If you want some extra credit, the function description reads that it _does not return any value_ - however, you are returning *a* value and the tests still pass.

Can you write a test in `word_processing_test.rb` that would fail `lower_case!` returns a value?

Reply:
`  def test_lower_case!
    words = ['', 'HELLO', 'WoRlD', 'nice']
    #lower_case!(words)
    assert_nil lower_case!(words)
    assert_equal ['', 'hello', 'world', 'nice'], words
  end`
Is this correct

Reply:
Yeah, that works